core/primitives/src/runtime/config_store.rs

```diff
@@ -0,0 +1,103 @@
+use include_dir::{include_dir, Dir};
+
+use crate::runtime::config::RuntimeConfig;
+use crate::types::Gas;
+use crate::types::ProtocolVersion;
+use std::collections::BTreeMap;
+use std::iter::FromIterator;
+use std::ops::Bound;
+use std::sync::Arc;
+
+/// Stores runtime config for each protocol version where it was updated.
+#[derive(Debug)]
+pub struct RuntimeConfigStore {
+    /// Maps protocol version to the config with possibly modified `max_gas_burnt_view` limit.
+    store: BTreeMap<ProtocolVersion, Arc<RuntimeConfig>>,
+}
+
+impl RuntimeConfigStore {
+    /// Constructs a new store.
+    ///
+    /// If `max_gas_burnt_view` is provided, the property in wasm limit
+    /// configuration will be adjusted to given value.
+    pub fn new(max_gas_burnt_view: Option<Gas>) -> Self {
+        let runtime_configs_dir: Dir = include_dir!("../../nearcore/res/runtime_configs");
+        Self {
+            store: BTreeMap::from_iter(runtime_configs_dir.files().iter().map(|file| {
+                let mut config: RuntimeConfig = serde_json::from_slice(file.contents()).unwrap();
+                if let Some(gas) = max_gas_burnt_view {
+                    config.wasm_config.limit_config.max_gas_burnt_view = gas;
+                }
+                (
+                    file.path().file_stem().unwrap().to_str().unwrap().parse().unwrap(),
+                    Arc::new(config),
+                )
+            })),
+        }
+    }
+
+    /// Returns a `RuntimeConfig` for the corresponding protocol version.
+    ///
+    /// Note that even if some old version is given as the argument, this may
+    /// still return configuration which differs from configuration found in
+    /// genesis file by the `max_gas_burnt_view` limit.
+    pub fn get_config(&self, protocol_version: ProtocolVersion) -> &Arc<RuntimeConfig> {
+        self.store
+            .range((Bound::Unbounded, Bound::Included(protocol_version)))
+            .next_back()
+            .unwrap_or_else(|| {
```

Comment:
I wonder whether it is better to return an option here.

Reply:
Good question! As a rule of thumb, I usually do like pushing `.unwraps` to the caller, but I think in this case it's better to unwrap in the callee. 

My understanding is that we have an invariant that we only care about protocol versions below the the current max, so it's fine to assume that we know config for the current protocol version. In other words, if we return an option here, there's no way any current or future caller would want to not unwrap.

But thinking about this more, perhaps it's better to make this case statically impossible? We can require in the constructor that we pass config for the zero'th version, and then there's always a version which we can return.

Reply:
I think it is fine to rename `29.json` to `0.json` and avoid panic, this should not break anything.